src/domain/entities/Task.ts

```diff
@@ -1,3 +1,58 @@
+/** @format */
+
+import { Guard, IGuardProps, Result } from "../../common/ErrorHandling";
 import { Entity } from "./Entity";
+import { ITaskProps } from "./interfaces";
+
+export class Task extends Entity<ITaskProps> {
+  private constructor({ id, ...data }: ITaskProps) {
+    super(data, id);
+  }
+
+  public static create(taskProps: ITaskProps) {
+    const guardedProps: IGuardProps[] = [
```

Comment:
you can also use validator, `zod` for example

Reply:
@mubin-khalid wouldn't that add dependency to the entity?